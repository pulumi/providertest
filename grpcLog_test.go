package pulumitest_test

import (
	"testing"

	"github.com/gkampitakis/go-snaps/match"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/pulumi/pulumitest"
	"github.com/stretchr/testify/assert"
)

func TestGrpc(t *testing.T) {
	t.Parallel()
	test := pulumitest.NewPulumiTest(t, "testdata/yaml_program")
	test.Up(t)
	log := test.GrpcLog(t)
	assert.NotEmpty(t, log)
	creates, err := log.Creates()
	assert.NoError(t, err)
	assert.Len(t, creates, 1)
	snaps.MatchJSON(t, creates, match.Any("0.Response.id", "0.Response.properties.id"))
}
