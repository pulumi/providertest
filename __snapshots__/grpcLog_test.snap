
[TestGrpc - 1]
[
 {
  "Request": {
   "name": "username",
   "properties": {},
   "type": "random:index/randomPet:RandomPet",
   "urn": "urn:pulumi:test::yaml_program::random:index/randomPet:RandomPet::username"
  },
  "Response": {
   "id": "<Any value>",
   "properties": {
    "id": "<Any value>",
    "length": 2,
    "separator": "-"
   }
  }
 }
]
---
